homeassistant/components/unifiprotect/binary_sensor.py

```diff
@@ -364,6 +367,24 @@ async def async_setup_entry(
 ) -> None:
     """Set up binary sensors for UniFi Protect integration."""
     data: ProtectData = hass.data[DOMAIN][entry.entry_id]
+
+    async def _add_new_device(device: ProtectAdoptableDeviceModel) -> None:
+        entities: list[ProtectDeviceEntity] = async_all_device_entities(
+            data,
+            ProtectDeviceBinarySensor,
+            camera_descs=CAMERA_SENSORS,
+            light_descs=LIGHT_SENSORS,
+            sense_descs=SENSE_SENSORS,
+            lock_descs=DOORLOCK_SENSORS,
+            viewer_descs=VIEWER_SENSORS,
+            ufp_device=device,
+        )
+        if device.is_adopted and isinstance(device, Camera):
+            entities += _async_motion_entities(data, ufp_device=device)
+        async_add_entities(entities)
+
+    async_dispatcher_connect(hass, _ufpd(entry, DISPATCH_ADOPT), _add_new_device)
```

Comment:
We need to disconnect the dispatcher signal when unloading the config entry. Please use `entry.async_on_unload`.